pkg/controller/common/keystore/user_secret.go

```diff
@@ -154,7 +154,40 @@ func retrieveUserSecret(c k8s.Client, recorder record.EventRecorder, hasKeystore
 	} else if err != nil {
 		return nil, false, err
 	}
-	return &userSecret, true, nil
+
+	// If no items, return the whole user secret
+	if secret.Items == nil {
```

Comment:
I was confused there for a second by the variable name `secret` because it actually is the `secretSrc` and we use `userSecret` for the actual secret.

Reply:
Agree, it can be confusing. `secretSrc` looks better.